config/initializers/faraday_patch.rb

```diff
@@ -0,0 +1,11 @@
+# Monkey patch https://github.com/lostisland/faraday/pull/513
+# Fixes encoding issue when passing an URL with non UTF-8 characters
+module Faraday
+  module Utils
+    def unescape(s)
+      string = s.to_s
+      string.force_encoding(Encoding::BINARY) if RUBY_VERSION >= '1.9'
```

Comment:
The idea of the faraday PR is to not care about the encoding of the URL at all, thus the conversion to binary. If we would replace all non UTF-8 characters the request would probably fail or return different results. The patch fixes the issue with the URL reported in #1588